elf_load.c

```diff
@@ -198,11 +198,20 @@ static int elf_load_parse(struct elf_prog *elf_prog, Elf *elf)
 				elf_prog->upperl = phdr.p_paddr + phdr.p_memsz;
 		}
 		UK_ASSERT(elf_prog->lowerl <= elf_prog->upperl);
+
+		if (phdr.p_offset <= ehdr.e_phoff &&
+		    ehdr.e_phoff < phdr.p_offset + phdr.p_filesz)
+			elf_prog->phdr.off = ehdr.e_phoff - phdr.p_offset +
+					     phdr.p_paddr;
```

Comment:
I noticed that here, and everywhere else in this file we use `phdr.p_paddr`. According to ELF(5):

> On systems for which physical addressing is relevant, this member is reserved for the segment's
>  physical address.  Under BSD this member is not used and must be zero.

Is there any reason we don't use `p_vaddr` instead? To me it also sounds semantically more appropriate as we are operating on the virtual address space.

Reply:
I agree! I have had the same concern as you when I first touched the code, but I've chosen to just follow what the other code did and still use `p_paddr`, But yes it's a valid concern :+1:. @skuenzer Any idea why the initial code uses `p_paddr` everywhere? Should there also be a commit changing it all to `p_vaddr` instead?

Reply:
😮 oh, oh. I guess we should change it. How and why does it work today?

Reply:
In my experience, despite this about paddr
> Under BSD this member is not used and must be zero.

I have yet to personally meet an ELF whose paddr != vaddr. So I would say yes they are equal.

> we should fix it with another PR or commit

sure, I can do this :+1: should be just a `s/p_paddr/p_vaddr`

Reply:
Yes, please. Thanks for finding.